ecc/secp256k1/ecdsa/ecdsa.go

```diff
@@ -110,6 +112,48 @@ func HashToInt(hash []byte) *big.Int {
 	return ret
 }
 
+// RecoverP recovers the value P (prover commitment) when creating a signature.
+// It uses the recovery information v and part of the decomposed signature r. It
+// is used internally for recovering the public key.
+func RecoverP(v uint, r *big.Int) (*secp256k1.G1Affine, error) {
+	x := new(big.Int).Set(r)
+	// if x is r or r+N
```

Comment:
is it a necessary check when cofactor=1?  both r and p are on 256 bits

Reply:
right but `P = [k]G` where `k` is a random generated by (`randFieldElement`). So the probability that the `x` coordinate of a randomly generated point falls between `r` and `p` is so tiny (should be around `10^{-39}`). But this doesn't impact perf nor code complexity I'm fine with it.

Reply:
So, there is this: https://coders-errand.com/malleability-ecdsa-signatures/, https://eklitzke.org/bitcoin-transaction-malleability

For MtGox hack because both (v, r, s) and (v, r+Fr, s) were valid signatures for the same message, then it was possible to double spend as only hashes of the signatures were checked.

Cost-wise, I think this method isn't much slower (we compute one additional mul). However now I see that for it to make sense we have to add comparison x < Fr

Reply:
I think MtGox was about `y/-y` if I'm not mistaken. But anyway it doesn't hurt to keep this.